src/lightning_lite/connector.py

```diff
@@ -0,0 +1,596 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from collections import Counter
+from typing import Dict, List, Optional, Union
+
+import torch
+
+from lightning_lite.accelerators import ACCELERATOR_REGISTRY
+from lightning_lite.accelerators.accelerator import Accelerator
+from lightning_lite.accelerators.cuda import CUDAAccelerator
+from lightning_lite.accelerators.mps import MPSAccelerator
+from lightning_lite.accelerators.tpu import TPUAccelerator
+from lightning_lite.plugins import (
+    CheckpointIO,
+    DeepSpeedPrecision,
+    NativeMixedPrecision,
+    Precision,
+    TPUBf16Precision,
+    TPUPrecision,
+)
+from lightning_lite.plugins.environments import (
+    ClusterEnvironment,
+    KubeflowEnvironment,
+    LightningEnvironment,
+    LSFEnvironment,
+    SLURMEnvironment,
+    TorchElasticEnvironment,
+)
+from lightning_lite.plugins.precision.double import DoublePrecision
+from lightning_lite.strategies import (
+    DDPShardedStrategy,
+    DDPSpawnShardedStrategy,
+    DDPSpawnStrategy,
+    DDPStrategy,
+    DeepSpeedStrategy,
+    SingleDeviceStrategy,
+    SingleTPUStrategy,
+    Strategy,
+    STRATEGY_REGISTRY,
+    TPUSpawnStrategy,
+)
+from lightning_lite.strategies.ddp_spawn import _DDP_FORK_ALIASES
+from lightning_lite.utilities import _StrategyType, device_parser, rank_zero_deprecation, rank_zero_info, rank_zero_warn
+from lightning_lite.utilities.imports import _HPU_AVAILABLE, _IPU_AVAILABLE, _IS_INTERACTIVE, _TPU_AVAILABLE
+
+_PLUGIN = Union[Strategy, Precision, ClusterEnvironment, CheckpointIO]
+_PLUGIN_INPUT = Union[_PLUGIN, str]
+
+
+class _Connector:
```

Comment:
why is this class protected? IMO it shouldn't be, just it's instance in Lite should be protected.

Reply:
My reasoning here was simply that we want to communicate this is not an internal class that a user should customize, override or use in anyway outside of Lite. I'm fine either way, @carmocca for second opinion.

Reply:
I'm not saying that they should be public in general, I'm just saying that they should be public w.r.t the namespace of this file.

Reply:
Since breaking changes and deprecations are such a large issue, I prefer to be safe and prefix everything that somebody could ever think of importing.

What you say makes some sense but it's impossible to communicate to the users other than through the documentation

Nowadays nobody looks at imports, as the IDE usually manages them. Nobody checks that they imported:
`from lightning_lite.connector import Connector` (supposedly protected externally, public internally)
instead of
`from lightning_lite import Connector` (public externally and internally)

Reply:
I do :D

but fine by me :)